### PR TITLE
fix deprecation notice from in IriConvertor::getIriFromItem

### PR DESCRIPTION
**Pull request by sh41**
Per [OperationTypeDeprecationHelper.php](https://github.com/api-platform/core/blob/42940e3f48c3450d6c193f6bca6b8c5041821ffa/src/Api/OperationTypeDeprecationHelper.php#L18) comment: `false` should be replaced with `OperationType::ITEM`. 

Per deprecation notice in [RouteNameResolver](https://github.com/api-platform/core/blob/e66ed60aa0ea4065f75cbd8c24ef07d8f2b14abb/src/Bridge/Symfony/Routing/RouteNameResolver.php#L44) third argument also becomes mandatory and is empty array when not in use. 

| Q             | A
| ------------- | ---
| Bug fix?      | yes - fix deprecation notice
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

I've tried, but I cannot figure out how to write a test to cover this change. The deprecation comes out of RouteNameResolver::getRouteName which has coverage, but the deprecated call is made from IriConverter::GetIriFromItem which doesn't appear to have coverage at the moment.

I ended up in a situation where I was attempting to mock/prophesize Router returning a mocked RouteCollection returning a mocked Route and began to suspect that I was following the wrong path. If anyone can provide pointers I'll be eternally grateful.

### CODE REVIEW

File: src/Bridge/Symfony/Routing/IriConverter.php [hunk 1]
```diff
@@ -92,7 +92,7 @@ public function getItemFromIri(string $iri, array $context = [])
     public function getIriFromItem($item, int $referenceType = UrlGeneratorInterface::ABS_PATH): string
     {
         $resourceClass = $this->getObjectClass($item);
-        $routeName = $this->routeNameResolver->getRouteName($resourceClass, false);
+        $routeName = $this->routeNameResolver->getRouteName($resourceClass, OperationType::ITEM, []);
```
Comment (soyuka) [inline src/Bridge/Symfony/Routing/IriConverter.php]: Hmm, this is supposed to work fine with false. It'll raise an exception but should not be blocking.

Comment (sh41) [inline src/Bridge/Symfony/Routing/IriConverter.php]: Yes, that's right - it functions perfectly well, but logs two deprecation notices. The only effect of the change is that the two deprecations notices are no longer logged.

Comment (meyerbaptiste) [inline src/Bridge/Symfony/Routing/IriConverter.php]: To prevent deprecation notices, you have to change these lines too:
* [IriConverter.php#L114](https://github.com/api-platform/core/blob/2139c802abd52a81af7f6f16132eea55ad081e14/src/Bridge/Symfony/Routing/IriConverter.php#L114)
* [IriConverter.php#L126](https://github.com/api-platform/core/blob/2139c802abd52a81af7f6f16132eea55ad081e14/src/Bridge/Symfony/Routing/IriConverter.php#L126)

But yes, it's a little weird to have a `trigger_error()` with an optional argument... 😄

Comment (soyuka) [inline src/Bridge/Symfony/Routing/IriConverter.php]: Hmm yeah you're right this last argument is optional. I added this because the interface will change but maybe it's not appropriate and could be silenced. WDYT?

Comment (meyerbaptiste) [inline src/Bridge/Symfony/Routing/IriConverter.php]: IMO we should because `Not defining it is deprecated since 2.1` and the `E_USER_DEPRECATED` are wrong if it's an optional argument.

Comment (soyuka) [inline src/Bridge/Symfony/Routing/IriConverter.php]: :+1:, @sh41 may you update the pull request by removing the `trigger_error` instead of changing files where it's being used? Thanks!

Comment (sh41) [inline src/Bridge/Symfony/Routing/IriConverter.php]: Thanks @soyuka I have removed the trigger error from both the CachedRouteNameResolver and  RouteNameResolver. 

I'm not familiar with the syntax of the optional argument in this interface/class - if there is no deprecation to handle can the method signatures also be refactored?

```php
public function getRouteName(string $resourceClass, $operationType /**, array $context = [] **/): string
```
changed to 
```php
public function getRouteName(string $resourceClass, $operationType , array $context = [] ): string
```
and drop this bit?
```php
        if (func_num_args() > 2) {
            $context = func_get_arg(2);
        } else {
            $context = [];
        }
```

In CachedRouteNameResolver, RouteNameResolver and RouteNameResolverInterface?

Comment (soyuka) [inline src/Bridge/Symfony/Routing/IriConverter.php]: Haha no we can't do this because it'd need to change the interface, which becomes then a breaking change (ie major version). Leave the hack alone, it'll be removed in 3.x (hopefully).

File: src/Bridge/Symfony/Routing/IriConverter.php [hunk 2]
```diff
@@ -92,7 +92,7 @@ public function getItemFromIri(string $iri, array $context = [])
     public function getIriFromItem($item, int $referenceType = UrlGeneratorInterface::ABS_PATH): string
     {
         $resourceClass = $this->getObjectClass($item);
-        $routeName = $this->routeNameResolver->getRouteName($resourceClass, false);
+        $routeName = $this->routeNameResolver->getRouteName($resourceClass, OperationType::ITEM);
```
Comment (soyuka) [inline src/Bridge/Symfony/Routing/IriConverter.php]: I don't think we need this change, anyway as both argument types are supported yet I'd say we can keep it like this.

Comment (sh41) [inline src/Bridge/Symfony/Routing/IriConverter.php]: There's a deprecation notice at [OperationTypeDeprecationHelper.php#L39](https://github.com/api-platform/core/blob/42940e3f48c3450d6c193f6bca6b8c5041821ffa/src/Api/OperationTypeDeprecationHelper.php#L39) that suggests that although the boolean type works at the moment it will stop working in v3. Happy to revert the change if it is premature.

Comment (meyerbaptiste) [inline src/Bridge/Symfony/Routing/IriConverter.php]: 👍 for this change, we should not use our own deprecated code.

Comment (soyuka) [inline src/Bridge/Symfony/Routing/IriConverter.php]: If we don't how would we now they're deprecated? ^^ Anyway, [there are a bunch of places where we still use `false`](https://github.com/api-platform/core/search?utf8=%E2%9C%93&q=getRouteName&type=), though we were going to fix those only when breaking.

Comment (meyerbaptiste) [inline src/Bridge/Symfony/Routing/IriConverter.php]: For me, deprecated notices should be for the users of our public API and should be avoided in our internals. And in our tests we should only have one for the deprecation test (i.e. `@group legacy` + `@expectedDeprecation`).

**Comment by soyuka** [comment]
thanks @sh41
